src/memphis.ts

```diff
@@ -168,18 +174,44 @@ export class Memphis {
             this.maxReconnect = maxReconnect > 9 ? 9 : maxReconnect;
             this.reconnectIntervalMs = reconnectIntervalMs;
             this.timeoutMs = timeoutMs;
-
             let conId_username = this.connectionId + '::' + username;
             try {
-                this.brokerManager = await broker.connect({
-                    servers: `${this.host}:${this.port}`,
-                    reconnect: this.reconnect,
-                    maxReconnectAttempts: this.reconnect ? this.maxReconnect : 0,
-                    reconnectTimeWait: this.reconnectIntervalMs,
-                    timeout: this.timeoutMs,
-                    token: this.connectionToken,
-                    name: conId_username
-                });
+                if (keyFile !== '' || certFile !== '' || caFile !== '') {
+                    if (keyFile === '') {
+                        return reject(MemphisError(new Error('Must provide a TLS key file')));
+                    }
+                    if (certFile === '') {
+                        return reject(MemphisError(new Error('Must provide a TLS cert file')));
+                    }
+                    if (caFile === '') {
+                        return reject(MemphisError(new Error('Must provide a TLS ca file')));
+                    }
+                    let tlsOptions = {
+                        keyFile: keyFile,
+                        certFile: certFile,
+                        caFile: caFile
+                    };
+                    this.brokerManager = await broker.connect({
```

Comment:
move the connection options into a variable and inject the tls options if needed instead of re-write it all again